packages/patternfly-3/patternfly-react/src/components/LoadingState/LoadingState.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Spinner } from '../Spinner';
+
+class LoadingState extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      render: false
+    };
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({ render: true });
+    }, this.props.timeout);
+  }
+
+  render() {
+    const { loading, loadingText, children } = this.props;
+    const spinner = (
+      <div className="pf-loading-state">
+        <Spinner loading={loading} size="lg" />
+        <p>{loadingText}</p>
+      </div>
+    );
+
+    if (loading) {
+      return this.state.render ? spinner : null;
+    }
+    return children;
+  }
+}
+LoadingState.propTypes = {
+  loading: PropTypes.bool,
+  loadingText: PropTypes.string,
+  children: PropTypes.node,
+  timeout: PropTypes.number
+};
```

Comment:
the propTypes need comments to add descriptions in storybook.

Reply:
This should take a size for the entire loading state (rather than `spinnerSize`) and add `loading-state-pf-[lg, sm, xs]` to the outer div and set the spinner size appropriately